### PR TITLE
GH-44101: [C++][Parquet] Tools: Debug Print for Json should be valid JSON

### PR DESCRIPTION
**Pull request by mapleFU**
### Rationale for this change

The printJson is not a valid json now. This is ok for human-read, but when I want to analysis it with json tools or ai, it will prevent from using it.

### What changes are included in this PR?

Change the output to be a valid json.

Style:

previously, the `\"` trailing would be added in start of object, but this patch put it to end of object

Before:
```
stream << "\", \"number\":\""  << number;
stream << "\"...";
```

After:

```
stream << ", \"number\":\""  << number << "\"";
```

### Are these changes tested?

Yes

### Are there any user-facing changes?

Minor format change

* GitHub Issue: #44101

### CODE REVIEW

**Comment by mapleFU** [comment]
@pitrou @wgtmac this is a minor change for tool with testing. Would you mind take a look?

File: cpp/src/parquet/reader_test.cc [hunk 1]
```diff
@@ -1172,6 +1180,30 @@ TEST_F(TestJSONWithLocalFile, JSONOutputFLBA) {
   EXPECT_THAT(json_content, testing::HasSubstr(json_contains));
 }
 
+// GH-44101: Test that JSON output is valid JSON
+TEST_F(TestJSONWithLocalFile, ValidJsonOutput) {
+  auto checkJsonValid = [](std::string_view json_string) -> ::arrow::Status {
```
Comment (pitrou) [inline cpp/src/parquet/reader_test.cc:1185]: Nit: `check_valid_json`?

**Review by pitrou — APPROVED** [review]

File: cpp/src/parquet/reader_test.cc (hunk 1 above)
Comment (mapleFU) [reply cpp/src/parquet/reader_test.cc]: done

**Comment by mapleFU** [comment]
https://github.com/apache/arrow/actions/runs/11514744258/job/32053979673?pr=44532

Emmm seems this is orc related, and unrelated to this patch? 🤔

**Comment by pitrou** [comment]
@mapleFU The error message seems related, perhaps we need to make RapidJSON mandatory when Parquet tests are enabled?
```
D:/a/arrow/arrow/cpp/src/parquet/reader_test.cc(32): fatal error C1083: Cannot open include file: 'rapidjson/document.h': No such file or directory
```

**Comment by mapleFU** [comment]
Aha, my bad, will fix this

File: cpp/src/parquet/CMakeLists.txt [hunk 2]
```diff
@@ -320,7 +320,7 @@ if(ARROW_TESTING)
   # "link" our dependencies so that include paths are configured
   # correctly
   target_link_libraries(parquet_testing PUBLIC ${ARROW_GTEST_GMOCK})
-  list(APPEND PARQUET_TEST_LINK_LIBS parquet_testing)
+  list(APPEND PARQUET_TEST_LINK_LIBS parquet_testing RapidJSON)
```
Comment (mapleFU) [inline cpp/src/parquet/CMakeLists.txt:323]: add rapidjson to `PARQUET_TEST_LINK_LIBS`, hope will not affect too much

**Review by wgtmac — APPROVED** [review]

**Comment by mapleFU** [comment]
Would merge this firstly

**Pull request merged**